eternalcore-core/src/main/java/com/eternalcode/core/feature/randomteleport/RandomTeleportCommand.java

```diff
@@ -25,17 +31,36 @@ class RandomTeleportCommand {
 
     private final NoticeService noticeService;
     private final RandomTeleportService randomTeleportService;
+    private final PluginConfiguration config;
+    private final Delay<UUID> delay;
 
     @Inject
-    RandomTeleportCommand(NoticeService noticeService, RandomTeleportService randomTeleportService) {
+    RandomTeleportCommand(NoticeService noticeService, RandomTeleportService randomTeleportService, PluginConfiguration config) {
         this.noticeService = noticeService;
         this.randomTeleportService = randomTeleportService;
+        this.config = config;
+        this.delay = new Delay<>(this.config.randomTeleport);
     }
 
     @Execute
     @Permission("eternalcore.rtp")
     @DescriptionDocs(description = "Teleportation of the sender to a random location.")
     void executeSelf(@Context Player player) {
+        UUID uuid = player.getUniqueId();
+
+        if (this.delay.hasDelay(uuid)) {
+            Duration time = this.delay.getDurationToExpire(uuid);
+
+            this.noticeService
+                .create()
+                .notice(translation -> translation.randomTeleport().randomTeleportDelay())
+                .placeholder("{TIME}", DurationUtil.format(time))
+                .player(uuid)
+                .send();
+
+            return;
+        }
```

Comment:
Since this code is repeated in 2 places, it should have a single method for checking whether the player is in cooldown.